modules/common/src/main/scala/protocol.scala

```diff
@@ -37,8 +33,7 @@ sealed abstract class CompressionType extends Product with Serializable
 case object Identity                  extends CompressionType
 case object Gzip                      extends CompressionType
 
-class rpc(val serializationType: SerializationType, val compressionType: CompressionType = Identity)
-    extends StaticAnnotation
+class rpc(val compressionType: CompressionType = Identity) extends StaticAnnotation
```

Comment:
I think that we could just have the compression type as another annotation, e.g. `@gzip`, instead of an argument of `@rpc`.